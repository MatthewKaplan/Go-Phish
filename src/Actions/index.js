export const allMembers = members => ({
  type: "ALL_MEMBERS",
  members
});

export const currentMember = member => ({
  type: "CURRENT_MEMBER",
  member
});

export const allYears = years => ({
  type: "ALL_YEARS",
  years
});

export const allTours = tours => ({
  type: "ALL_TOURS",
  tours
});

export const allSongs = songs => ({
  type: "ALL_SONGS",
  songs
});

export const allVenues = venues => ({
  type: "ALL_VENUES",
  venues
});

export const currentShows = shows => ({
  type: "CURRENT_SHOWS",
  shows
});
