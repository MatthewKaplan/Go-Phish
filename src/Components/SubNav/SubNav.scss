.sub-nav {
  div {
    cursor: pointer;
  }
}