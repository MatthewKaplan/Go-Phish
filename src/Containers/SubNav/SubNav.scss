.sub-nav {
  color: #6e6f6f;
  text-decoration: none;
  div {
    cursor: pointer;
  }

  .char_link {
    color: red;
    text-decoration: none;
  }
}
