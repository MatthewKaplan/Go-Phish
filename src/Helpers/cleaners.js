import React from "react";

const cleanSongs = data => {
  const songInfo = data.map(song => {
    return {
      id: song.id,
      slug: song.slug,
      title: song.title,
      tracks_count: song.tracks_count
    };
  });
  return songInfo;
};

const cleanVenues = data => {
  const venueInfo = data.map(venue => {
    return {
      country: venue.country,
      id: venue.id,
      location: venue.location,
      name: venue.name,
      show_dates: venue.show_dates,
      show_ids: venue.show_ids,
      shows_count: venue.shows_count,
      slug: venue.slug
    };
  });
  return venueInfo;
};

const cleanTours = data => {
  const tourInfo = data.map(tour => {
    return {
      ends_on: tour.ends_on,
      id: tour.id,
      name: tour.name,
      shows_count: tour.shows_count,
      slug: tour.slug,
      starts_on: tour.starts_on,
      shows: tour.shows.map(show => {
        return {
          date: show.date,
          id: show.id,
          location: show.location,
          tour_id: show.tour_id,
          venue_id: show.venue_id,
          venue_name: show.venue_name
        };
      })
    };
  });
  return tourInfo;
};

const cleanShows = data => {
  const showInfo = data.map(show => {
    return {
      date: show.date,
      id: show.id,
      tour_id: show.tour_id,
      venue_name: show.venue_name,
      location: show.venue.location,
      tracks: show.tracks.map(track => {
        return {
          id: track.id,
          mp3: track.mp3,
          position: track.position,
          set_name: track.set_name,
          show_id: track.show_id,
          title: track.title,
          song_ids: track.song_ids[0]
        };
      }),
      venue: show.venue
    };
  });
  return showInfo;
};

// const cleanTourShows = data => {
//   const showInfo = data.map(show => {
//     return {
//       date: show.date,
//       id: show.id,
//       tour_id: show.tour_id,
//       venue_name: show.venue_name,

//     }
//   })
// }

export { cleanSongs, cleanVenues, cleanTours, cleanShows };
