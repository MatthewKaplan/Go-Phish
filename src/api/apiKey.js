const bearer_token =
  "47db3d11c8dd5d03ac33a54db9fa07efeb5afd643c581445c15ac5d8e158fce1b96e7d845e8dbecd67c97280185ca938";
export const bearer = "Bearer " + bearer_token;
