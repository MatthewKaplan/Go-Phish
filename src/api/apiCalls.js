import { bearer } from "./apiKey";
const url = "https://cors-anywhere.herokuapp.com/http://phish.in/api/v1/";

const fetchData = path => {
  return fetch(url + path, {
    method: "GET",
    headers: {
      Authorization: bearer,
      Accept: "application/json",
      "Content-Type": "application/json"
    }
  }).then(response => {
    if (!response.ok) {
      throw new Error("Fetch failed");
    } else {
      return response.json();
    }
  });
};

const fetchMembers = path => {
  return fetch(path).then(response => {
    if (!response.ok) {
      throw new Error("Fetch failed");
    } else {
      return response.json();
    }
  });
};

export { fetchData, fetchMembers };
