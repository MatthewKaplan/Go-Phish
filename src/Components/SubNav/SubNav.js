import React, { Component } from "react";
import { fetchData } from "../../api/apiCalls";
import { connect } from "react-redux";
import { allSongs, allVenues, handleError } from "../../Actions/index";
import "./SubNav.scss";
import { cleanSongs, cleanVenues } from "../../Helpers/cleaners";
import PropTypes from "prop-types";

export class SubNav extends Component {
  state = { venues: [], songs: [], searchChar: "A" };
  componentDidMount() {
    this.fetchVenues();
    this.fetchSongs();
  }

  fetchVenues = async () => {
    try {
      const venuesResults = await fetchData(`venues.json?per_page=651`);
      const cleanVenuesResults = await cleanVenues(venuesResults.data);
      await this.setState({ venues: cleanVenuesResults });
    } catch (error) {
      this.props.handleError(error.message);
    }
  };

  fetchSongs = async () => {
    try {
      const songsResults = await fetchData(`songs.json?per_page=901`);
      const cleanSongsResults = await cleanSongs(songsResults.data);
      await this.setState({ songs: cleanSongsResults });
    } catch (error) {
      this.props.handleError(error.message);
    }
  };

  filterVenues = async path => {
    await this.props.allVenues(this.state.venues);
    this.toggleActiveState(path);
    const { venues } = this.props;
    const filteredVenues = venues.filter(venue => {
      return venue.name[0] === path;
    });
    this.props.allVenues(filteredVenues);
  };

  filterSongs = path => {
    this.props.allSongs(this.state.songs);
    this.toggleActiveState(path);
    const { songs } = this.props;
    const filteredSongs = songs.filter(song => {
      return song.title[0] === path;
    });
    this.props.allSongs(filteredSongs);
  };

  toggleActiveState = char => {
    this.setState({
      searchChar: char
    });
  };

  whatToFetch = (char, path) => {
    if (path === "http://localhost:3000/Venues") {
      this.filterVenues(char);
    } else {
      this.filterSongs(char);
    }
  };

  render() {
    const { searchChar } = this.state;
    return (
      <nav className="sub-nav">
        <div
          className={searchChar === "A" ? "char_link" : null}
          data-test="a-btn"
          onClick={() => this.whatToFetch("A", window.location.href)}
        >
          A
        </div>
        <div
          className={searchChar === "B" ? "char_link" : null}
          data-test="b-btn"
          onClick={() => this.whatToFetch("B", window.location.href)}
        >
          B
        </div>
        <div
          className={searchChar === "C" ? "char_link" : null}
          data-test="c-btn"
          onClick={() => this.whatToFetch("C", window.location.href)}
        >
          C
        </div>
        <div
          className={searchChar === "D" ? "char_link" : null}
          data-test="d-btn"
          onClick={() => this.whatToFetch("D", window.location.href)}
        >
          D
        </div>
        <div
          className={searchChar === "E" ? "char_link" : null}
          data-test="e-btn"
          onClick={() => this.whatToFetch("E", window.location.href)}
        >
          E
        </div>
        <div
          className={searchChar === "F" ? "char_link" : null}
          data-test="f-btn"
          onClick={() => this.whatToFetch("F", window.location.href)}
        >
          F
        </div>
        <div
          className={searchChar === "G" ? "char_link" : null}
          data-test="g-btn"
          onClick={() => this.whatToFetch("G", window.location.href)}
        >
          G
        </div>
        <div
          className={searchChar === "H" ? "char_link" : null}
          data-test="h-btn"
          onClick={() => this.whatToFetch("H", window.location.href)}
        >
          H
        </div>
        <div
          className={searchChar === "I" ? "char_link" : null}
          data-test="i-btn"
          onClick={() => this.whatToFetch("I", window.location.href)}
        >
          I
        </div>
        <div
          className={searchChar === "J" ? "char_link" : null}
          data-test="j-btn"
          onClick={() => this.whatToFetch("J", window.location.href)}
        >
          J
        </div>
        <div
          className={searchChar === "K" ? "char_link" : null}
          data-test="k-btn"
          onClick={() => this.whatToFetch("K", window.location.href)}
        >
          K
        </div>
        <div
          className={searchChar === "L" ? "char_link" : null}
          data-test="l-btn"
          onClick={() => this.whatToFetch("L", window.location.href)}
        >
          L
        </div>
        <div
          className={searchChar === "M" ? "char_link" : null}
          data-test="m-btn"
          onClick={() => this.whatToFetch("M", window.location.href)}
        >
          M
        </div>
        <div
          className={searchChar === "N" ? "char_link" : null}
          data-test="n-btn"
          onClick={() => this.whatToFetch("N", window.location.href)}
        >
          N
        </div>
        <div
          className={searchChar === "O" ? "char_link" : null}
          data-test="o-btn"
          onClick={() => this.whatToFetch("O", window.location.href)}
        >
          O
        </div>
        <div
          className={searchChar === "P" ? "char_link" : null}
          data-test="p-btn"
          onClick={() => this.whatToFetch("P", window.location.href)}
        >
          P
        </div>
        <div
          className={searchChar === "Q" ? "char_link" : null}
          data-test="q-btn"
          onClick={() => this.whatToFetch("Q", window.location.href)}
        >
          Q
        </div>
        <div
          className={searchChar === "R" ? "char_link" : null}
          data-test="r-btn"
          onClick={() => this.whatToFetch("R", window.location.href)}
        >
          R
        </div>
        <div
          className={searchChar === "S" ? "char_link" : null}
          data-test="s-btn"
          onClick={() => this.whatToFetch("S", window.location.href)}
        >
          S
        </div>
        <div
          className={searchChar === "T" ? "char_link" : null}
          data-test="t-btn"
          onClick={() => this.whatToFetch("T", window.location.href)}
        >
          T
        </div>
        <div
          className={searchChar === "U" ? "char_link" : null}
          data-test="u-btn"
          onClick={() => this.whatToFetch("U", window.location.href)}
        >
          U
        </div>
        <div
          className={searchChar === "V" ? "char_link" : null}
          data-test="v-btn"
          onClick={() => this.whatToFetch("V", window.location.href)}
        >
          V
        </div>
        <div
          className={searchChar === "W" ? "char_link" : null}
          data-test="w-btn"
          onClick={() => this.whatToFetch("W", window.location.href)}
        >
          W
        </div>
        <div
          className={searchChar === "X" ? "char_link" : null}
          data-test="x-btn"
          onClick={() => this.whatToFetch("X", window.location.href)}
        >
          X
        </div>
        <div
          className={searchChar === "Y" ? "char_link" : null}
          data-test="y-btn"
          onClick={() => this.whatToFetch("Y", window.location.href)}
        >
          Y
        </div>
        <div
          className={searchChar === "Z" ? "char_link" : null}
          data-test="z-btn"
          onClick={() => this.whatToFetch("Z", window.location.href)}
        >
          Z
        </div>
      </nav>
    );
  }
}

SubNav.propTypes = {
  songs: PropTypes.array,
  venues: PropTypes.array,
  allSongs: PropTypes.func,
  allVenues: PropTypes.func,
  handleError: PropTypes.func
};

export const mapStateToProps = state => ({
  songs: state.songs,
  venues: state.venues
});

export const mapDispatchToProps = dispatch => ({
  allSongs: songs => dispatch(allSongs(songs)),
  allVenues: venues => dispatch(allVenues(venues)),
  handleError: errorMessage => dispatch(handleError(errorMessage))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SubNav);
