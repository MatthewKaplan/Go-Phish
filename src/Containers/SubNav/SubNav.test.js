import React from "react";
import { shallow } from "enzyme";
import * as actions from "../../Actions/index";
import MockData from "../../Helpers/mockData";
import { SubNav, mapStateToProps, mapDispatchToProps } from "./SubNav";
import { fetchData } from "../../api/apiCalls";
import { cleanVenues, cleanSongs } from "../../Helpers/cleaners";

jest.mock("../../api/apiCalls.js");
jest.mock("../../Helpers/cleaners.js");

const mockSong = MockData.mockSong;
const mockVenuesResults = MockData.mockVenue;
const mockVenues = MockData.mockVenues;
const mockUncleanVenues = MockData.mockUncleanVenues;
const mockCleanVenues = jest.fn();
const mockAllVenues = jest.fn();
const mockAllSongs = jest.fn();
const mockToggleActiveState = jest.fn();
const mockWhatToFetch = jest.fn();
const mockHandleError = jest.fn();

describe("SubNav", () => {
  let wrapper, instance;
  fetchData.mockImplementation(() => Promise.resolve(1));

  beforeEach(() => {
    wrapper = shallow(
      <SubNav
        songs={mockSong}
        venues={mockVenues}
        cleanVenues={mockCleanVenues}
        allVenues={mockAllVenues}
        allSongs={mockAllSongs}
        toggleActiveState={mockToggleActiveState}
        whatToFetch={mockWhatToFetch}
        venuesResults={mockVenuesResults}
        handleError={mockHandleError}
      />
    );
    instance = wrapper.instance();
  });

  afterEach(() => {
    fetchData.mockClear();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have default state", () => {
    wrapper = shallow(
      <SubNav
        songs={mockSong}
        venues={mockVenues}
        cleanVenues={mockCleanVenues}
      />
    );
    expect(wrapper.state()).toEqual({
      venues: [],
      songs: [],
      searchChar: "A"
    });
  });

  it("should invoke 'fetchVenues, fetchSongs' when mounted", () => {
    jest.spyOn(instance, "fetchVenues");
    jest.spyOn(instance, "fetchSongs");
    instance.componentDidMount();
    expect(instance.fetchSongs).toHaveBeenCalled();
    expect(instance.fetchVenues).toHaveBeenCalled();
  });

  describe("fetchVenues", () => {
    it("should invoke 'fetchData' with correct params when 'fetchVenues' is invoked", () => {
      instance.fetchVenues();
      expect(fetchData).toHaveBeenCalledWith("venues.json?per_page=651");
    });

    it("should clean the response of 'fetchVenues' when invoked", async () => {
      cleanVenues.mockImplementation(() => Promise.resolve(mockUncleanVenues));
      instance.fetchVenues();
      expect(cleanVenues).toHaveBeenCalled();
    });

    it("should set the state of 'venues' after data is cleaned", async () => {
      await instance.fetchVenues();
      expect(wrapper.state().venues).toEqual(mockUncleanVenues);
    });

    it("should finally throw an error if the response is not ok and save that error to redux store", async () => {
      fetchData.mockImplementationOnce(() =>
        Promise.reject(new Error("Fetch failed"))
      );
      await wrapper.instance().fetchVenues();
      expect(mockHandleError).toHaveBeenCalledWith("Fetch failed");
    });
  });

  describe("fetchSongs", () => {
    it("should invoke 'fetchData' with correct params when 'fetchSongs' is invoked", () => {
      instance.fetchSongs();
      expect(fetchData).toHaveBeenCalledWith("songs.json?per_page=901");
    });

    it("should clean the response of 'fetchSongs' when invoked", async () => {
      cleanSongs.mockImplementation(() => Promise.resolve(mockSong));
      instance.fetchSongs();
      expect(cleanSongs).toHaveBeenCalled();
    });

    it("should set the state of 'songs' after data is cleaned", async () => {
      await instance.fetchSongs();
      expect(wrapper.state().songs).toEqual(mockSong);
    });

    it("should finally throw an error if the response is not ok and save that error to redux store", async () => {
      fetchData.mockImplementationOnce(() =>
        Promise.reject(new Error("Fetch failed"))
      );
      await wrapper.instance().fetchSongs();
      expect(mockHandleError).toHaveBeenCalledWith("Fetch failed");
    });
  });

  describe("toggleActiveState", () => {
    it("should change the state of 'searchChar' to the character passed into it when invoked", () => {
      expect(wrapper.state().searchChar).toEqual("A");
      instance.toggleActiveState("Z");
      expect(wrapper.state().searchChar).toEqual("Z");
    });
  });

  describe("filterVenues", () => {
    it("should invoke 'allVenues' with the correct params when invoked", () => {
      instance.filterVenues("M");
      expect(mockAllVenues).toHaveBeenCalledWith([]);
    });
  });

  describe("filterSongs", () => {
    it("should invoke 'allSongs' with the correct params when invoked", () => {
      instance.filterSongs("M");
      expect(mockAllSongs).toHaveBeenCalledWith([]);
    });
  });

  describe("whatToFetch", () => {
    it("should invoke 'filterVenues' with the correct params", () => {
      jest.spyOn(instance, "filterVenues");
      instance.whatToFetch("A", "http://localhost:3000/Venues");
      expect(instance.filterVenues).toHaveBeenCalledWith("A")
    })

    it("should invoke 'filterSongs' with the correct params", () => {
      jest.spyOn(instance, "filterSongs");
      instance.whatToFetch("A", "http://localhost:3000/Songs");
      expect(instance.filterSongs).toHaveBeenCalledWith("A")
    })
  })

  describe("Nav Clicks", () => {
    beforeEach(() => {
      jest.spyOn(instance, "whatToFetch");
    });

    it("should invoke 'whatToFetch' with the param 'A' ", () => {
      wrapper.find("[data-test='a-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("A", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'B' ", () => {
      wrapper.find("[data-test='b-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("B", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'C' ", () => {
      wrapper.find("[data-test='c-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("C", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'D' ", () => {
      wrapper.find("[data-test='d-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("D", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'E' ", () => {
      wrapper.find("[data-test='e-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("E", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'F' ", () => {
      wrapper.find("[data-test='f-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("F", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'G' ", () => {
      wrapper.find("[data-test='g-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("G", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'H' ", () => {
      wrapper.find("[data-test='h-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("H", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'I' ", () => {
      wrapper.find("[data-test='i-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("I", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'J' ", () => {
      wrapper.find("[data-test='j-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("J", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'K' ", () => {
      wrapper.find("[data-test='k-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("K", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'L' ", () => {
      wrapper.find("[data-test='l-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("L", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'M' ", () => {
      wrapper.find("[data-test='m-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("M", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'N' ", () => {
      wrapper.find("[data-test='n-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("N", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'O' ", () => {
      wrapper.find("[data-test='o-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("O", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'P' ", () => {
      wrapper.find("[data-test='p-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("P", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'Q' ", () => {
      wrapper.find("[data-test='q-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("Q", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'R' ", () => {
      wrapper.find("[data-test='r-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("R", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'S' ", () => {
      wrapper.find("[data-test='s-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("S", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'T' ", () => {
      wrapper.find("[data-test='t-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("T", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'U' ", () => {
      wrapper.find("[data-test='u-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("U", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'V' ", () => {
      wrapper.find("[data-test='v-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("V", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'W' ", () => {
      wrapper.find("[data-test='w-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("W", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'X' ", () => {
      wrapper.find("[data-test='x-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("X", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'Y' ", () => {
      wrapper.find("[data-test='y-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("Y", window.location.href);
    });

    it("should invoke 'whatToFetch' with the param 'Z' ", () => {
      wrapper.find("[data-test='z-btn']").simulate("click");
      expect(instance.whatToFetch).toHaveBeenCalledWith("Z", window.location.href);
    });
  });
});

describe("mapStateToProps", () => {
  it("should return an object", () => {
    const mockData = {
      songs: [],
      venues: []
    };
    const expected = {
      songs: [],
      venues: []
    };

    const mockprops = mapStateToProps(mockData);
    expect(mockprops).toEqual(expected);
  });
});

describe("mapDispatchToProps", () => {
  it("should call dispatch for allSongs", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.allSongs(mockSong);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.allSongs(mockSong);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should call dispatch for allVenues", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.allVenues(mockVenues);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.allVenues(mockVenues);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should call dispatch for handleError", () => {
    const error = "Mock Error";
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.handleError(error);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleError(error);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
