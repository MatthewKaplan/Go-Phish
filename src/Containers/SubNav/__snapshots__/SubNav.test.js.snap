// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNav should match the snapshot 1`] = `ShallowWrapper {}`;
